projects/noloan2/app/src/main/java/hasadna/noloan2/MainActivity.java

```diff
@@ -67,11 +80,44 @@ private FirestoreElement encodeUser(loguser user) {
   @Override
   public void onClick(View v) {
     switch (v.getId()) {
-      case R.id.B_adduser:
+      case R.id.B_adduser: // Clicking on the add user button
         loguser user = loguser.newBuilder()
           .setName(name.getText().toString())
           .build();
         writeUser(user);
     }
   }
+  
+  // The user can change the permission after the splash screen, so checking for permission is still needed.
```

Comment:
The this comment should probably replace `Check for permission reading sms` below.
Generally, method-level comments provide a high-level explanation of the method.
I'd write something like:
`\\ Reads SMS. If no permissions are granted, exit app.`